recipes/archicad-apidevkit/all/conanfile.py

```diff
@@ -0,0 +1,69 @@
+from conan import ConanFile
+from conan.tools.files import get, copy, save
+from conan.errors import ConanInvalidConfiguration
+import textwrap
+import os
+
+required_conan_version = ">=1.46.0"
+
+
+class ArchicadApidevkitConan (ConanFile):
+    name = "archicad-apidevkit"
+    description = "The General API Development Kit enables software developers to extend the functionality of Archicad"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://archicadapi.graphisoft.com/"
+    license = "LicenseRef-LICENSE"
+    settings = "os", "arch", "build_type", "compiler"
+    no_copy_source = True
+    topics = "api, Archicad, development"
+    short_paths = True
+
+    @property
+    def _acdevkit_arch(self):
+        return str(self.settings.arch)
+
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
+
```

Comment:
```suggestion

    def validate(self):
        if not self.info.settings.os in ("Macos", "Windows"):
            raise ConanInvalidConfiguration(f"{self.ref} is not supported by the OS {self.info.settings.os}")

```
Yes, we will need to forced validate because the CI will run linux by default.